File: RockitCompiler/benchmarks/bench_field_extract.cpp
#include <cstdio>
#include <string>

int main() {
    std::string line = "f0";
    for (int i = 1; i < 100; i++) {
        line += ",f";
        line += std::to_string(i);
    }

    std::string result;
    for (int iter = 0; iter < 500000; iter++) {
        int fieldIdx = 0;
        size_t start = 0;
        for (size_t j = 0; j < line.size(); j++) {
            if (line[j] == ',') {
                if (fieldIdx == 50) {
                    result = line.substr(start, j - start);
                    break;
                }
                fieldIdx++;
                start = j + 1;
            }
        }
    }
    printf("%s\n", result.c_str());
    return 0;
}

File: RockitCompiler/benchmarks/bench_field_extract.rs
fn main() {
    let mut line = String::from("f0");
    for i in 1..100 {
        line.push(',');
        line.push('f');
        line.push_str(&i.to_string());
    }

    let mut result = String::new();
    for _ in 0..500_000 {
        let mut field_idx = 0;
        let mut start = 0;
        let bytes = line.as_bytes();
        for j in 0..bytes.len() {
            if bytes[j] == b',' {
                if field_idx == 50 {
                    result = line[start..j].to_string();
                    break;
                }
                field_idx += 1;
                start = j + 1;
            }
        }
    }
    println!("{}", result);
}
